server/game/cards/01_SOR/DeathTrooper.ts

```diff
@@ -18,13 +18,15 @@ export default class DeathTrooper extends NonLeaderUnitCard {
                     cardTypeFilter: WildcardCardType.Unit,
                     controller: RelativePlayer.Self,
                     locationFilter: Location.GroundArena,
-                    immediateEffect: AbilityHelper.immediateEffects.damage({ amount: 2 })
+                    immediateEffect: AbilityHelper.immediateEffects.damage({ amount: 2 }),
+                    optional: true
```

Comment:
This ability isn't optional. I saw in the PR description that there was some special handling for this, but I feel like this is really unclear/unintuitive to be setting to true here. Is there more to it?

Reply:
The target that must be fulfilled if possible but the ability still happens if not possible. Naming that property is tricky; the best I can think of is canResolveWithout.

Reply:
Discussed in chat